src/netbase.cpp

```diff
@@ -605,10 +588,7 @@ bool ConnectSocketByName(CService &addr, SOCKET& hSocketRet, const char *pszDest
 
     SplitHostPort(string(pszDest), port, strDest);
 
-    proxyType nameProxy;
-    GetNameProxy(nameProxy);
-
-    CService addrResolved(CNetAddr(strDest, fNameLookup && !HaveNameProxy()), port);
+    CService addrResolved(CNetAddr(strDest, fNameLookup && !HaveProxy(NET_IPV4)), port);
```

Comment:
Could completely bypass this "lookup" in the case that a proxy is used, after all we always pass the name to the proxy. e.g.
```
ProxyType proxy;
if(GetProxy(NET_NAME, proxy))
{
    addr = CService("0.0.0.0:0"); 
    return ConnectThroughProxy(proxy, strDest, port, hSocketRet, nTimeout, outProxyConnectionFailed);
}
```
This can even further simplify the code.